src/relayFeeCalculator/relayFeeCalculator.ts

```diff
@@ -0,0 +1,97 @@
+import assert from "assert";
+import * as uma from "@uma/sdk";
+import { Provider } from "@ethersproject/providers";
+import { BigNumber } from "ethers";
+import { BigNumberish, calculateGasFees, toBNWei } from "../utils";
+const Coingecko = uma.Coingecko;
+const { percent, fixedPointAdjustment } = uma.across.utils;
+const { connect: erc20Connect } = uma.clients.erc20;
+
+export interface QueryInterface {
+  getGas: (tokenAddress?: string) => Promise<BigNumberish>;
+  getGasPrice: () => Promise<BigNumberish>;
+  getTokenPrice: (tokenAddress: string) => Promise<number | string>;
+  getTokenDecimals: (tokenAddress: string) => Promise<number>;
+}
+
+// Override this to add different behaviors for different chains.
+export class DefaultQueries implements QueryInterface {
+  private coingecko: uma.Coingecko;
+  constructor(private provider: Provider, private priceSymbol: string = "eth", private defaultGas = "305572") {
+    this.coingecko = new Coingecko();
+  }
+  async getGas(): Promise<BigNumberish> {
```

Comment:
let me try to work this in and see how it changes things, but yea im for that if it makes this more flexible. 

Theres some assumptions made, that the fees require a gas amount, gas price in native token, and then a conversion price from gas token to relayed token.  it sounds like some chains may break those assumptions.

Reply:
i think the matic polygon case is fine with this interface, the optimism case sounds like it would be difficult. need to know more about how this works. 

for the matic case, we would need to override `getTokenPrice` and pass it in the token address based on the matic chain, and it would need to convert the token to its price in matic. 

for the optimsim case, it sounds like you need to get 2 different gas estimations and 2 different token prices to be able to convert to those tokens

Reply:
i've updated the interface to combine the gas calculations.